src/deploy.ts

```diff
@@ -83,6 +83,36 @@ export class HypERC20Deployer extends GasRouterDeployer<
     }
   }
 
+  // Gets the metadata for a collateral token, favoring the config
+  // and getting any on-chain metadata that is missing.
+  async getCollateralMetadata(
+    chain: ChainName,
+    config: CollateralConfig,
+  ): Promise<ERC20Metadata> {
+    const metadata = {
+      name: config.name,
+      symbol: config.symbol,
+      decimals: config.decimals,
+      totalSupply: 0,
+    };
+
+    if (metadata.name && metadata.symbol && metadata.decimals !== undefined && metadata.decimals !== null) {
+      return metadata as ERC20Metadata;
+    }
+    const fetchedMetadata = await HypERC20Deployer.fetchMetadata(
+      this.multiProvider.getProvider(chain),
+      config,
+    );
+    // Filter out undefined values
+    const definedConfigMetadata = Object.fromEntries(
+      Object.entries(metadata).filter(([_, v]) => v !== undefined)
+    );
+    return {
+      ...fetchedMetadata,
+      ...definedConfigMetadata,
+    } as ERC20Metadata;
```

Comment:
iiuc the spread and merge will do the undefined filter automatically?

Reply:
Hmm, I don't think it will if a property is `definedConfigMetadata` has been explicitly set to undefined or null

Reply:
learned this the hard way, I thought the same as you Yorke, but the tests caught this

Reply:
woof js is hard